From 251f503ba02a2ca364c8ff46f805def4663c9010 Mon Sep 17 00:00:00 2001
From: Jordan Hendricks <jordan.hendricks@joyent.com>
Date: Wed, 24 May 2017 22:03:50 +0000
Subject: [PATCH] joyent/node-manta#308 `mmpu commit` does not parse options
 joyent/node-manta#309 MPU tests are out of sync with Muskie master branch
 implementation Reviewed by: Richard Kiene <richard.kiene@joyent.com> Approved
 by: Richard Kiene <richard.kiene@joyent.com>

---
 bin/mmpu            | 9 ++++++++-
 lib/client.js       | 4 ----
 test/client.test.js | 8 ++++----
 test/mmpu.test.js   | 8 ++++----
 4 files changed, 16 insertions(+), 13 deletions(-)

diff --git a/bin/mmpu b/bin/mmpu
index 640d8bb..7f27650 100755
--- a/bin/mmpu
+++ b/bin/mmpu
@@ -222,6 +222,7 @@ MMpu.prototype.do_create.help = [
     '',
     'Usage:',
     '    mmpu create [OPTIONS] PATH',
+    '',
     '{{options}}',
     ''
 ].join('\n');
@@ -433,6 +434,9 @@ MMpu.prototype.do_commit = function do_commit(subcmd, opts, args, cb) {
 };
 
 
+MMpu.prototype.do_commit.options = manta.DEFAULT_CLI_OPTIONS;
+
+
 MMpu.prototype.do_commit.help = [
     'Commit a multipart upload.',
     '',
@@ -470,6 +474,8 @@ MMpu.prototype.do_commit.help = [
     '',
     'Usage:',
     '    mmpu commit [OPTIONS] UPLOAD_ID [ETAG_0] [ETAG_1] ...',
+    '',
+    '{{options}}',
     ''
 ].join('\n');
 
@@ -634,7 +640,8 @@ MMpu.prototype.do_parts.help = [
     '',
     'Usage:',
     '    mmpu parts [OPTIONS] UPLOAD_ID',
-    ''
+    '',
+    '{{options}}'
 ].join('\n');
 
 
diff --git a/lib/client.js b/lib/client.js
index 5772c69..a029d1d 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -3361,7 +3361,6 @@ MantaClient.prototype.createUpload = function createUpload(p, opts, cb) {
 
         self.jsonClient.post(options, body, function (err2, req, res, obj) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient POST');
                 log.debug(err2);
                 cb(err2);
             } else {
@@ -3486,7 +3485,6 @@ MantaClient.prototype.abortUpload = function abortUpload(id, opts, cb) {
 
         self.jsonClient.post(options, function (err2, req, res, obj) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient POST');
                 log.debug(err2);
                 cb(err2);
             } else {
@@ -3539,7 +3537,6 @@ MantaClient.prototype.getUpload = function getUpload(id, opts, cb) {
 
         self.jsonClient.get(options, function (err2, req, res, obj) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient GET');
                 log.debug(err2);
                 cb(err2);
             } else {
@@ -3593,7 +3590,6 @@ MantaClient.prototype.commitUpload = function commitUpload(id, p, opts, cb) {
         var body = { parts:  p };
         self.jsonClient.post(options, body, function (err2, req, res) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient POST');
                 log.debug(err2);
                 cb(err2);
             } else {
diff --git a/test/client.test.js b/test/client.test.js
index 3136dd1..29fd7d6 100644
--- a/test/client.test.js
+++ b/test/client.test.js
@@ -791,8 +791,8 @@ test('commit upload', function (t) {
             t.ifError(err2);
             t.ok(upload);
             t.equal(upload.id, UPLOAD1);
-            t.equal(upload.state, 'finalizing');
-            t.equal(upload.type, 'commit');
+            t.equal(upload.state, 'done');
+            t.equal(upload.result, 'committed');
 
             self.client.get(PATH1, function (err3, stream) {
                 t.ifError(err3);
@@ -834,8 +834,8 @@ test('abort upload', function (t) {
                 t.ifError(err3);
                 t.ok(upload);
                 t.equal(upload.id, UPLOAD2);
-                t.equal(upload.state, 'finalizing');
-                t.equal(upload.type, 'abort');
+                t.equal(upload.state, 'done');
+                t.equal(upload.result, 'aborted');
 
                 t.done();
             });
diff --git a/test/mmpu.test.js b/test/mmpu.test.js
index 2a69312..c5be2e5 100644
--- a/test/mmpu.test.js
+++ b/test/mmpu.test.js
@@ -508,8 +508,8 @@ test('mmpu commit C_ID C_ETAG0', function (t) {
                 cb(err);
             } else {
                 var upload = JSON.parse(info.stdout);
-                t.ok(upload.state, 'finalizing');
-                t.ok(upload.type, 'commit');
+                t.ok(upload.state, 'done');
+                t.ok(upload.result, 'committed');
                 cb();
             }
         });
@@ -585,8 +585,8 @@ test('mmpu abort A_ID', function (t) {
                 cb(err);
             } else {
                 var upload = JSON.parse(info.stdout);
-                t.ok(upload.state, 'finalizing');
-                t.ok(upload.type, 'abort');
+                t.ok(upload.state, 'done');
+                t.ok(upload.result, 'aborted');
                 cb();
             }
         });
-- 
2.21.0

